packages/core/src/runtime.ts

```diff
@@ -371,11 +373,13 @@ export class AgentRuntime implements IAgentRuntime {
 
         this.token = opts.token;
 
-        this.plugins = [
+        const allPlugins = [
             ...(opts.character?.plugins ?? []),
             ...(opts.plugins ?? []),
         ];
 
+        this.plugins = _.uniqBy(allPlugins, "name");
```

Comment:
could this be done with a Set and array.from?

Reply:
Yes, I have changed it to that now @odilitime